redfish-core/lib/cable.hpp

```diff
@@ -60,17 +99,236 @@ inline void
+        [asyncResp](const boost::system::error_code& ec,
+                    const dbus::utility::DBusPropertiesMap& properties) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("DBUS response error {}", ec.value());
+            messages::internalError(asyncResp->res);
+            return;
+        }
+
+        afterFillCableProperties(asyncResp, properties);
+    });
+}
+
+inline void
+    fillCableHealthState(const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+                         const std::string& cableObjectPath,
+                         const std::string& service)
+{
+    sdbusplus::asio::getProperty<bool>(
+        *crow::connections::systemBus, service, cableObjectPath,
+        "xyz.openbmc_project.Inventory.Item", "Present",
+        [asyncResp, cableObjectPath](const boost::system::error_code& ec,
+                                     bool present) {
+        if (ec)
+        {
+            BMCWEB_LOG_DEBUG("get presence failed for Cable {} with error {}",
+                             cableObjectPath, ec.value());
+            if (ec.value() != EBADR)
+            {
+                messages::internalError(asyncResp->res);
+            }
+            return;
+        }
+
+        if (!present)
+        {
+            asyncResp->res.jsonValue["Status"]["State"] = "Absent";
+        }
+    });
+}
+
+inline void getCableUpstreamResources(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& cableObjectPath)
+{
+    // retrieve Upstream Resources
+    std::string upstreamResource = cableObjectPath + "/upstream_resource";
+    linkAssociatedCable(
+        asyncResp, upstreamResource,
+        [asyncResp](const dbus::utility::MapperEndPoints& endpoints) {
+        nlohmann::json::array_t linkArray;
+        for (const auto& fullPath : endpoints)
+        {
+            sdbusplus::message::object_path path(fullPath);
+            std::string leaf = path.filename();
+            if (leaf.empty())
+            {
+                continue;
+            }
+            nlohmann::json::object_t item;
+            item["@odata.id"] = boost::urls::format(
+                "/redfish/v1/Systems/system/PCIeDevices/{}", leaf);
+            linkArray.emplace_back(item);
+        }
+        asyncResp->res.jsonValue["Links"]["UpstreamResources@odata.count"] =
+            linkArray.size();
+        asyncResp->res.jsonValue["Links"]["UpstreamResources"] =
+            std::move(linkArray);
+    });
+}
+
+inline void doGetCableDownstreamResources(
+    const std::shared_ptr<bmcweb::AsyncResp>& asyncResp,
+    const std::string& cableObjectPath,
+    const std::vector<std::string>& updatedAssemblyList)
+{
+    std::string downstreamResource = cableObjectPath + "/downstream_resource";
+    linkAssociatedCable(asyncResp, downstreamResource,
+                        [asyncResp, cableObjectPath, updatedAssemblyList](
+                            const dbus::utility::MapperEndPoints& endpoints) {
```

Comment:
Should this > 10 line lambda be in a named function?

Reply:
I refactored those lambda usage cases.